server.js

```diff
@@ -11,37 +11,68 @@ let mongoStream;
-    }
+    const changeStream = mongoStream.changeStreams[changeStreams[i]];
+    if (changeStream)
+      responseBody[changeStream.collection] = 'Listening';
+    else
+      responseBody[changeStreams[i]] = 'Not Listening';
   }
 
   response.send(responseBody);
 });
 
-// triggers a manual collection dump for the specified collection
-app.get('/dump/:collection', (request, response) => {
-  console.log(`dumping collection ${request.params.collection} to ES`);
-  mongoStream.collectionDump(request.params.collection);
-  response.send(`dumping collection ${request.params.collection} to ES`);
+// triggers an add for the specified collections
+// @param collections: comma-separated string of collections to operate on
+// @param filter(optional): if exclusive, exclude the defined collections, else include them
+app.get('/add/:collections/:filter?', (request, response) => {
+  mongoStream.filterCollections({
+    filterArray: request.params.collections.split(','),
+    filterType: request.params.filter
+  }).then((collections) => {
+      return mongoStream.addChangeStreams(collections)
+    })
+    .then((results) => {
+      response.send(results);
+    });
+
 });
 
-// starts listening to the change stream of the specified collection
-app.get('/start/:collection', (request, response) => {
-  mongoStream.addChangeStream(request.params.collection).then(() => {
-    response.send(`${request.params.collection} change stream added`);
-  });
+// triggers a remove for the specified collections
+// @param collections: comma-separated string of collections to operate on
+// @param filter(optional): if exclusive, exclude the defined collections, else include them
+app.get('/remove/:collections/:filter?', (request, response) => {
+  mongoStream.filterCollections({
+    filterArray: request.params.collections.split(','),
+    filterType: request.params.filter
+  }).then((collections) => {
+    return mongoStream.removeChangeStreams(collections)
+  })
+    .then((results) => {
+      response.send(results);
+    });
 });
 
-// stops listening to the change stream of the specified collection
-app.get('/stop/:collection', (request, response) => {
-  mongoStream.removeChangeStream(request.params.collection);
-  response.send(`${request.params.collection} change stream removed`);
+// triggers a dump for the specified collections
+// @param collections: comma-separated string of collections to operate on
+// @param filter(optional): if exclusive, exclude the defined collections, else include them
+app.get('/dump/:collections/:filter?', (request, response) => {
+  mongoStream.filterCollections({
+    filterArray: request.params.collections.split(','),
+    filterType: request.params.filter
+  }).then((collections) => {
+    return mongoStream.dumpCollections(collections)
+  })
+    .then((results) => {
+      response.send(results);
+    });
+});
+
+// manually set the bulk size for replication testing
+app.get('/bulk=:bulkSize', (request, response) => {
```

Comment:
This is interesting. It is technically idempotent, but mutates global state. Probably better as a POST.

Reply:
I tend to prefer reserving `GET`s for truly read-only operations. Idempotency is more critical when distinguishing between `POST` and `PUT`.

Reply:
Thus `PUT` makes sense here (to me).

Reply:
For reference: https://en.wikipedia.org/wiki/Idempotence

Reply:
I always thought PUT is best when you are creating a resource at the resource path in question, not mutating the resource. But... since it is wholesale replacing the resource that makes a lot of sense. Talked myself into it.